demo/auth/passport.js

```diff
@@ -0,0 +1,47 @@
+var $fh = require('fh-js-sdk');
+var logger = require('@raincatcher/logger').logger;
+var ConsoleLogger = require('@raincatcher/logger').ConsoleLogger;
+var setLogger = require('@raincatcher/logger').setLogger;
+
+$fh.init({}, function() {
+  var passport = {
+    loadUserProfile: function($http, $window) {
+      var req = {
+        method: 'GET',
+        url: $fh.getCloudURL() + '/profile'
+      };
+      return $http(req, {withCredentials: true}).then(function (res) {
+        return res.data;
+      }, function (err) {
+        if (err.status === 401) {
+          $window.location = $fh.getCloudURL() + '/login';
+        }
+        if (err.status === 403) {
+          logger.error('Forbidden');
+        }
+        return err;
+      });
+    },
+    hasResourceRole: function(role) {
+      // TODO
+    },
+    logout: function($http, $window) {
+      var req = {
+        method: 'GET',
+        url: $fh.getCloudURL() + '/logout'
+      };
+
+      return $http(req, {withCredentials: true}).then(function(res) {
+        $window.location = $fh.getCloudURL() + '/login';
+      }, function(err) {
+        logger.error('An error occurred when logging out', err);
```

Comment:
I think we can change these logs to use cordova dialog on a separate ticket later on. +1 on the logger message. +1 in regards to the logout operation offline for stopping sync, etc. However login screen will be unavailable offline as this is hosted on the server.